src/websocket/worker.ts

```diff
@@ -45,6 +47,23 @@ async function initializeWebsocket(url, protocol) {
   })
 }
 
+function connect(timeout = 1000) {
+  setTimeout(function() {
+    initializeWebsocket().then(function() {
+      console.log('connect')
+      self.postMessage({
+        type: 'WEBSOCKET_OPEN',
+      })
+      ws.addEventListener('message', onWebsocketMessage)
+      ws.addEventListener('error', onWebsocketError)
+      ws.addEventListener('close', onWebsocketClose)
```

Comment:
When is this close event emitted?

Reply:
When server or browser close the connection.

Reply:
Yes, but when does that happen? Or I guess that it can happen for many reasons.. For example if the network connection is interrupted, is it `error` or `close` event?

Reply:
it can happen for many reasons. If close event caused by client,  then it will remove event listeners and close the connection in all other cases it will try to reconnect.